spring-ai-core/src/main/java/org/springframework/ai/chat/prompt/ChatOptions.java

```diff
@@ -25,14 +25,8 @@ public interface ChatOptions extends ModelOptions {
 
 	Float getTemperature();
 
-	void setTemperature(Float temperature);
```

Comment:
removing these setters also required to remove the `@Override` annotations on all implementing classes.  As a result the project didn't compile.  Please make sure the changes you make compile before sending a PR.